### DISCUSSION TITLE
Is there any equivalent Metasploit Script that can generate the following output

### DISCUSSION THREAD

**Discussion by flamecopper** [rapid7/metasploit-framework#16234] [Q&A] [answered]
https://nmap.org/nsedoc/scripts/smb2-security-mode.html
| smb2-security-mode:
|   3.1.1:
|_    Message signing enabled but not required

**Comment by bcoles** [comment, marked as answer]
The `auxiliary/scanner/smb/smb_version` module performs some fingerpinting and feature detection. This includes a check for SMB signing. If message signing is disabled the module should output `(signatures:optional)`.

https://github.com/rapid7/metasploit-framework/blob/f3228b4af7cfa4046313f86673f00096abb7f3ef/modules/auxiliary/scanner/smb/smb_version.rb#L195-L213
